client/controller/controller.js

```diff
@@ -0,0 +1,355 @@
+                    lng: address.geometry.location.lng()
+                  };
+
+                var service = new google.maps.places.PlacesService(map.gMap);
+
+                var request = {
+                  location: location,
+                 // placeId: restaurant.place_id,
+                  type: ['restaurant'],
+                  radius: '10000'
+                };
+
+                service.nearbySearch(
+                 request,
+                 function(place, status) {
+                  if (status === google.maps.places.PlacesServiceStatus.OK) {
+
+                    //filterdata is an array of buisnesses.
+                    $scope.filterdata = filterdata.filter(place);
+
+
+
+                    //this will be used for the reset button
+                    $scope.resetR = $scope.filterdata;
+
+                    //call the getdetail method.
+                    $scope.getDetails();
+
+
+              }
+            })
+
+
+
+          }
+        });
+
+
+
+
+    } //end of geocode function
+
+
+    $scope.getDetails = function(){
+
+       document.getElementById('map-canvas').style.visibility="visible";
+      document.getElementById('slideshow').style.visibility = "visible";
+
+
+      var service = new google.maps.places.PlacesService(map.gMap);
+      var listOfRestaurants = $scope.filterdata;
+
+
+      var id = listOfRestaurants[0].place_id;
+
+      console.log(listOfRestaurants[0]);
+
+        service.getDetails({
+          placeId: id
+        }, function(result, status){
+          if(status === google.maps.places.PlacesServiceStatus.OK){
+            console.log('results below');
+            console.log(result);
+
+
+            $scope.restaurantName = result.name;
+
+            $scope.callServer();
+
+            //display website
+            $scope.website        =  result.website;
+
+
+            if(result.reviews){
+              console.log('reviews are ' + result.reviews.length);
+               //document.getElementById('accordion').style.visibility = "visible";
+              //store reviews in scope variable
+              $scope.reviews      = result.reviews;
+
+
+            }
+
+
+          if(result.photos){
+            $scope.listOfPhotos = [];
+            for(var i=0; i < result.photos.length; i++){
+              var photos = result.photos[i].getUrl({'maxWidth': 600, 'maxHeight': 600});
+
+             $scope.listOfPhotos.push(photos);
+             //$scope.test = $scope.photos
+
+            }
+          }
+          else{
+            document.getElementById('slideshow').style.visibility= "hidden";
+            $scope.noPhotoText = "No photos to display"
+          }
+
+
+
+
+          } //end of service if
+
+
+        })
+
+
+    } //end of getDetail function.
+
+
+
+      //<-----------This is the function for the next button-------->
+    $scope.next = function(){
+
+
+
+      $scope.listOfPhotos = [];
+
+      var restaurant = $scope.filterdata;
+
+      $scope.reset();
```

Comment:
Use a generic function like `reset();` which does all the dirty work for you - eg. clearing array etc.